client/src/app/gateways/action-worker-watch/action-worker-watch.service.ts

```diff
@@ -0,0 +1,228 @@
+        }
+        this.subscribeToWorker(id);
+        let worker: ViewActionWorker;
+        try {
+            worker = await this.getFinishedWorker(id, watchActivity, true);
+        } catch (e) {
+            throw new Error(_(`Client has stopped watching worker: `) + actionName + ` (${e.message})`);
+        }
+        this.dialogService.removeAllDates(id);
+        this.unsubscribeFromWorker(id);
+        if (worker.state === ActionWorkerState.end) {
+            if (worker.result?.success === false) {
+                throw new HttpErrorResponse({
+                    error: { ...worker.result },
+                    url: originalResponse.url,
+                    status: worker.result?.status_code,
+                    statusText: worker.result?.message
+                });
+            } else {
+                return new HttpResponse<T>({
+                    body: worker.result,
+                    headers: originalResponse.headers,
+                    status: worker.result?.status_code,
+                    url: originalResponse.url,
+                    statusText: worker.result?.message
+                });
+            }
+        }
+        throw new HttpErrorResponse({
+            error: {
+                success: false,
+                message: `${worker.name} aborted without any specific message`,
+                url: originalResponse.url
+            },
+            status: worker.result.status_code,
+            statusText: worker.result?.message
+        });
+    }
+
+    /**
+     * Stops the service from showing a wait dialog for a specific action if it could not be written, is slow, or is inactive.
+     * Should only be used if the option to manage the handling of these specific action workers is done in a separate component.
+     * @param actionName the (backend) name of the action for which there shouldn't be a dialog.
+     */
+    public disableWaitDialog(actionName: string) {
+        if (!this._noDialogActionNames.find(name => name === actionName)) {
+            this._noDialogActionNames.push(actionName);
+        }
+    }
+
+    /**
+     * May be used to reverse the effects of enableWaitDialog.
+     */
+    public enableWaitDialog(actionName: string) {
+        const index = this._noDialogActionNames.findIndex(name => name === actionName);
+        if (!(index === -1)) {
+            this._noDialogActionNames.splice(index, 1);
+        }
+    }
+
+    /**
+     * Removes a worker from the currentWorkerIds.
+     * May be called from outside this service, to force the client to stop watching a running worker.
+     * If this worker is currently watched, the running watch command will throw an error "Client has stopped watching worker: <Action name>".
+     */
+    public async unsubscribeFromWorker(workerId: Id): Promise<void> {
+        this._currentWorkerIds = this._currentWorkerIds.filter(id => id !== workerId);
+        await this.refreshAutoupdateSubscription([workerId]);
```

Comment:
It might be a better idea to not update the au subscription on unsubscribe but only mark it for deletion, because it reopens the connection. For example in case multiple workers get resolved quickly one after another this would stress the au service unnecessarily. 

The subscriptions marked for deletion could be processed when a new worker is subscribed or all workers got resolved.